code/game/objects/items/weapons/cards_ids.dm

```diff
@@ -187,15 +187,23 @@
 
 /obj/item/weapon/card/id/New()
 	..()
-	spawn(30)
-	if(istype(loc, /mob/living/carbon/human))
-		blood_type = loc:dna:b_type
-		dna_hash = loc:dna:unique_enzymes
-		fingerprint_hash = md5(loc:dna:uni_identity)
+
+	spawn(30) //AWFULNESS AHOY
+	if(src && ishuman(loc))

```

Comment:
Well, last time I checked the ID card correctly waited for 3 seconds before grabbing its owner's info